### PR TITLE
Add interface for custom networkConfig

### PR DESCRIPTION
**Pull request by cybertron**
This provides a way to specify a custom networkConfig for each node
in a deployment. The user sets NETWORK_CONFIG_FOLDER to a path that
contains YAML files named for each node. The contents of these files
are included in the install-config definition for the appropriate
node.

### CODE REVIEW

**Comment by cybertron** [comment]
/hold

This is only a partial implementation (for static IPs we need a way to add DNS entries too), but I wanted to get it pushed as an example of what I was thinking for a generic networkConfig interface.

**Comment by cybertron** [comment]
/cc @creydr 
/hold cancel

This is working for me locally. We do need to sync with @derekhiggins to see if we can implement his static ip patch using this mechanism.

I tested with this network config directory:
```
==> network-config-static/hosts.yaml <==
- ip: 192.168.111.110
  hostnames:
    - "master-0"
- ip: 192.168.111.111
  hostnames:
    - "master-1"
- ip: 192.168.111.112
  hostnames:
    - "master-2"
- ip: 192.168.111.113
  hostnames:
    - "worker-0"
- ip: 192.168.111.114
  hostnames:
    - "worker-1"

==> network-config-static/ostest-master-0.yaml <==
networkConfig:
  interfaces:
  - name: enp2s0
    type: ethernet
    state: up
    ipv4:
      address:
      - ip: "192.168.111.110"
        prefix-length: 24
      enabled: true
  dns-resolver:
    config:
      server:
      - 192.168.111.1
  routes:
    config:
    - destination: 0.0.0.0/0
      next-hop-address: 192.168.111.1
      next-hop-interface: enp2s0

==> network-config-static/ostest-master-1.yaml <==
networkConfig:
  interfaces:
  - name: enp2s0
    type: ethernet
    state: up
    ipv4:
      address:
      - ip: "192.168.111.111"
        prefix-length: 24
      enabled: true
  dns-resolver:
    config:
      server:
      - 192.168.111.1
  routes:
    config:
    - destination: 0.0.0.0/0
      next-hop-address: 192.168.111.1
      next-hop-interface: enp2s0

==> network-config-static/ostest-master-2.yaml <==
networkConfig:
  interfaces:
  - name: enp2s0
    type: ethernet
    state: up
    ipv4:
      address:
      - ip: "192.168.111.112"
        prefix-length: 24
      enabled: true
  dns-resolver:
    config:
      server:
      - 192.168.111.1
  routes:
    config:
    - destination: 0.0.0.0/0
      next-hop-address: 192.168.111.1
      next-hop-interface: enp2s0

==> network-config-static/ostest-worker-0.yaml <==
networkConfig:
  interfaces:
  - name: enp2s0
    type: ethernet
    state: up
    ipv4:
      address:
      - ip: "192.168.111.113"
        prefix-length: 24
      enabled: true
  dns-resolver:
    config:
      server:
      - 192.168.111.1
  routes:
    config:
    - destination: 0.0.0.0/0
      next-hop-address: 192.168.111.1
      next-hop-interface: enp2s0

==> network-config-static/ostest-worker-1.yaml <==
networkConfig:
  interfaces:
  - name: enp2s0
    type: ethernet
    state: up
    ipv4:
      address:
      - ip: "192.168.111.114"
        prefix-length: 24
      enabled: true
  dns-resolver:
    config:
      server:
      - 192.168.111.1
  routes:
    config:
    - destination: 0.0.0.0/0
      next-hop-address: 192.168.111.1
      next-hop-interface: enp2s0```

**Comment by cybertron** [comment]
This now includes all of the changes needed to deploy with bonds. It still needs documentation, but I wanted to get feedback on whether the approach is okay before I do that work.

**Comment by cybertron** [comment]
/hold

I randomly discovered that this breaks the additional networks functionality needed for kubernetes-nmstate testing. I need to figure out how to make those play nicely before this goes in.

**Comment by cybertron** [comment]
/hold cancel

This version plays nice with extra networks too.

**Comment by tdomnesc** [comment]
I tried this on my dev server with: static IPs , static IPs with vlan, and also interface.ipv4.dhcp: true without a hosts.yaml file. All worked.

**Comment by cybertron** [comment]
Since it seems like people are happy with this I added documentation and made the folder path a little more user-friendly. I think this should be good to go.

File: network.sh [hunk 1]
```diff
@@ -188,6 +188,11 @@ if  [[ ! -z "${INSTALLER_PROXY:-}" ]]; then
   fi
 fi
 
+if [ -n "$NETWORK_CONFIG_FOLDER" ]; then
```
Comment (tdomnesc) [inline network.sh:191]: `network.sh: line 191: NETWORK_CONFIG_FOLDER: unbound variable
make: *** [Makefile:42: ironic] Error 1
`
04_setup_ironic.sh has "set -u" and it sources network.sh
replace with `"${NETWORK_CONFIG_FOLDER:-}"`

Comment (cybertron) [reply network.sh]: Oops, thanks. Fixed.

**Pull request merged**